lib/config.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+var Promise = require("bluebird");
+var Confidence = require("confidence");
+var Joi = require("joi");
+
+var configSettings;
+
+function Config() {
+	configSettings = {
+		bandwidth : {
+			userId : process.env.BW_USER_ID,
+			token  : process.env.BW_TOKEN,
+			secret : process.env.BW_SECRET
+		}
+	};
+	this.store = new Confidence.Store(configSettings);
+}
+
+Config.prototype.validate = function () {
+	var schema = Joi.object().keys({
+		bandwidth : Joi.object().keys({
+			userId : Joi.string().required(),
+			token  : Joi.string().required(),
+			secret : Joi.string().required()
+		}),
+	});
+
+	var validatePromise = Promise.promisify(Joi.validate);
```

Comment:
`Joi.validate` is synchronous when no callback is provided. See the code examples [here](https://github.com/hapijs/joi#validatevalue-schema-options-callback).